pkg/vpa/service.go

```diff
@@ -149,6 +149,32 @@ func (c *Service) CreateTortoiseUpdaterVPA(ctx context.Context, tortoise *autosc
 	return vpa, tortoise, nil
 }
 
+// UpdateVPAContainerResourcePolicy is update VPAs to have appropriate container policies based on tortoises' resource policy.
+func (c *Service) UpdateVPAContainerResourcePolicy(ctx context.Context, tortoise *autoscalingv1beta3.Tortoise, vpa *v1.VerticalPodAutoscaler) (*v1.VerticalPodAutoscaler, *autoscalingv1beta3.Tortoise, error) {
+	retVPA := &v1.VerticalPodAutoscaler{}
+	var err error
+	// we only want to record metric once in every reconcile loop.
+	//metricsRecorded := false
+	updateFn := func() error {
+		crp := make([]v1.ContainerResourcePolicy, 0, len(tortoise.Spec.ResourcePolicy))
+		for _, c := range tortoise.Spec.ResourcePolicy {
+			crp = append(crp, v1.ContainerResourcePolicy{
+				ContainerName: c.ContainerName,
+				MinAllowed:    c.MinAllocatedResources,
+			})
+		}
+		vpa.Spec.ResourcePolicy = &v1.PodResourcePolicy{ContainerPolicies: crp}
+		retVPA, err = c.c.AutoscalingV1().VerticalPodAutoscalers(vpa.Namespace).UpdateStatus(ctx, vpa, metav1.UpdateOptions{})
```

Comment:
There are two update methods, `Update` and `UpdateStatus` - the former is a general update function, whereas the latter is the update function for the `status` subresource. Given ResourcePolicy is a `spec` we should use a general `Update` method.

```suggestion
		retVPA, err = c.c.AutoscalingV1().VerticalPodAutoscalers(vpa.Namespace).Update(ctx, vpa, metav1.UpdateOptions{})
```